### PR TITLE
Docs: Implement SideNavigation

### PR DESCRIPTION
**Pull request by AlbertCarreras**
#### Why?

Docs: Implement SideNavigation

### Links

- [Jira](https://jira.pinadmin.com/browse/GESTALT-XXXX)
- [TDD](link to Paper doc)
- [Figma](link to Figma file)

### Checklist

- [ ] Added unit and Flow Tests
- [ ] Added documentation + accessibility tests
- [ ] Verified accessibility: keyboard & screen reader interaction
- [ ] Checked dark mode, responsiveness, and right-to-left support
- [ ] Checked stakeholder feedback (e.g. Gestalt designers)

### CODE REVIEW

File: docs/components/Navigation.js [hunk 1]
```diff
@@ -21,80 +19,80 @@ function getAlphabetizedComponents() {
...
-      ) : (
-        <Box marginTop={4}>
-          {getAlphabetizedComponents().map((componentName, i) => (
-            <SidebarSectionLink componentName={componentName} key={`${sidebarOrganisedBy}-${i}`} />
-          ))}
-        </Box>
-      )}
-    </Fragment>
+    <SideNavigation
+      accessibilityLabel="Side navigation"
```
Comment (ayeshakmaz) [inline docs/components/Navigation.js:28]: Maybe "Page navigation"?

File: docs/components/Navigation.js [hunk 2]
```diff
@@ -21,80 +19,80 @@ function getAlphabetizedComponents() {
...
-        ))
-      ) : (
-        <Box marginTop={4}>
-          {getAlphabetizedComponents().map((componentName, i) => (
-            <SidebarSectionLink componentName={componentName} key={`${sidebarOrganisedBy}-${i}`} />
-          ))}
-        </Box>
-      )}
-    </Fragment>
+    <SideNavigation
```
Comment (ayeshakmaz) [inline docs/components/Navigation.js:27]: I think we're missing the border

File: docs/components/Navigation.js [hunk 3]
```diff
@@ -21,80 +19,80 @@ function getAlphabetizedComponents() {
...
+      accessibilityLabel="Side navigation"
+      header={
+        <nav label="Sidenav sort">
+          <SidebarCategorizationButton
```
Comment (ayeshakmaz) [inline docs/components/Navigation.js:31]: Could use a little extra top padding here above this button, it's a bit tight with the header. Maybe just surround in a box with some top margin

Comment (AlbertCarreras) [reply docs/components/Navigation.js:31]: @ayeshakmaz Let me check with hector, cos we discussed the top margin here.

Comment (AlbertCarreras) [reply docs/components/Navigation.js:31]: DONE

File: docs/components/Navigation.js (hunk 2 above)
Comment (AlbertCarreras) [reply docs/components/Navigation.js:27]: DONE

**Review by ayeshakmaz — APPROVED** [review]

**Pull request merged**